src/utils/array-utils.js

```diff
@@ -0,0 +1,42 @@
+export function swapItemsAt(items, index1, index2) {
+  const lastIndex = items.length - 1;
+  if (index1 < 0 || index2 < 0 || index1 > lastIndex || index2 > lastIndex || index1 === index2) {
+    return items;
+  }
+
+  const result = items.slice();
+
+  const item1 = result[index1];
+  const item2 = result[index2];
+  result[index1] = item2;
```

Comment:
or: 
result[index1] = items[index2]
result[index2] = items[index1]
No need for the extra variables